src/test/java/dev/sigstore/rekor/client/RekorClientTest.java

```diff
@@ -26,20 +30,137 @@
+    var resp = client.putEntry(newRekordRequest);
+    assertEquals(
+        1,
+        client
+            .searchEntry(
+                null, newRekordRequest.getHashedrekord().getData().getHash().getValue(), null, null)
+            .size());
+  }
+
+  @Test
+  public void searchEntries_oneResult_publicKey()
+      throws IOException, CertificateException, NoSuchAlgorithmException, SignatureException,
+          URISyntaxException, InvalidKeyException, OperatorCreationException {
+    var newRekordRequest = createdRekorRequest();
+    var resp = client.putEntry(newRekordRequest);
+    assertEquals(
+        1,
+        client
+            .searchEntry(
+                null,
+                null,
+                "x509",
+                resp.getEntry().getBodyAsHashedrekord().getSignature().getPublicKey().getContent())
+            .size());
+  }
+
+  @Test
+  public void searchEntries_moreThanOneResult_email()
+      throws IOException, CertificateException, NoSuchAlgorithmException, SignatureException,
+          URISyntaxException, InvalidKeyException, OperatorCreationException {
+    var newRekordRequest = createdRekorRequest();
+    var newRekordRequest2 = createdRekorRequest();
+    client.putEntry(newRekordRequest);
+    client.putEntry(newRekordRequest2);
+    assertTrue(
+        client.searchEntry("test@test.com", null, null, null).size()
+            > 1); // as long as our tests use staging this is just going to grow.
+  }
+
+  @Test
+  public void searchEntries_zeroResults() throws IOException {
+    assertTrue(
+        client
+            .searchEntry(
+                null,
+                "sha256:9f54fad117567ab4c2c6738beef765f7c362550534ffc0bfe8d96b0236d69661", // made
+                // up sha
+                null,
+                null)
+            .isEmpty());
+  }
+
+  @Test
+  public void getEntry_entryExists()
+      throws IOException, CertificateException, NoSuchAlgorithmException, SignatureException,
+          URISyntaxException, InvalidKeyException, OperatorCreationException {
+    var newRekordRequest = createdRekorRequest();
+    var resp = client.putEntry(newRekordRequest);
+    var entry = client.getEntry(resp.getUuid());
+    assertTrue(entry.isPresent());
+    assertEquals(resp.getEntry().getLogID(), entry.get().getLogID());
+  }
+
+  @Test
+  public void getEntry_entryDoesntExist() throws IOException {
+    Optional<RekorEntry> entry =
+        client.getEntry(
+            "c8d2b213aa7efc1b2c9ccfa2fa647d00b34c63972e04e90276b5c31e0f317afd"); // I made this up
+    assertTrue(entry.isEmpty());
+  }
+
+  @NotNull
+  private HashedRekordRequest createdRekorRequest()
+      throws NoSuchAlgorithmException, InvalidKeyException, SignatureException,
+          OperatorCreationException, CertificateException, IOException, URISyntaxException {
     // the data we want to sign
-    var data = "some data";
+    var data = "some data " + System.currentTimeMillis();
```

Comment:
you can also use UUID.randomUUID() but I dunno if it matters.

Reply:
done.